File: app/models/group.rb
class Group < ActiveRecord::Base
  belongs_to :founder, class_name: "User", foreign_key: "user_id"
  has_many :memberships
  has_many :members, through: :memberships, source: :user

  after_create :add_founder_as_member

  def add_founder_as_member
    self.memberships.create(user_id: self.user_id)
  end
end

File: app/assets/javascripts/models/group.js
// for more details see: http://emberjs.com/guides/models/defining-models/

App.Group = DS.Model.extend({
  name: DS.attr('string'),
  private: DS.attr('boolean'),
  founder: DS.belongsTo('user'),
  memberships: DS.hasMany('membership'),

  members: function() {
    var memberships = this.get('memberships');
    var members = memberships.getEach('user');
    return members ;
  }.property('memberships'),

  memberCount: function() {
    return this.get('members.length');
  }.property('members')

});
